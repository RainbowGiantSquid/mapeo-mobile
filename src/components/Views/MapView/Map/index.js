// @flow
import { connect } from 'react-redux';
import { NavigationActions } from 'react-navigation';
import type { Dispatch } from 'redux';
import {
  observationList,
  observationCreate,
  observationUpdate,
  observationSelect
} from '../../../../ducks/observations';

import { StoreState } from '../../../../types/redux';
import Map from './Map';
import type { DispatchProps, StateProps } from './Map';

const mapStateToProps = (state: StoreState): StateProps => ({
  observations: state.app.observations,
  selectedObservation: state.app.selectedObservation,
  gps: state.app.gps.data
});

function mapDispatchToProps(dispatch: Dispatch): DispatchProps {
  return {
    listObservations: () => dispatch(observationList('')),
    createObservation: observation => dispatch(observationCreate(observation)),
    goToCategories: () =>
      dispatch(
        NavigationActions.navigate({
          routeName: 'Categories',
          key: 'CategoriesView'
        })
      ),
    updateObservation: observation => dispatch(observationUpdate(observation)),
    selectObservation: o => dispatch(observationSelect(o)),
    goToObservationDetail: () =>
      dispatch(
        NavigationActions.navigate({ routeName: 'ObservationDetailView' })
      )
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Map);
