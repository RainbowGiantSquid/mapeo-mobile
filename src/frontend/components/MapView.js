// @flow
import React from "react";
import MapboxGL from "@mapbox/react-native-mapbox-gl";

import type { MapStyle } from "../types/map";
import type { ObservationsMap } from "../context/ObservationsContext";
import AddButton from "../components/AddButton";
import debug from "debug";

const log = debug("mapeo:MapView");

const mapboxStyles = MapboxGL.StyleSheet.create({
  observation: {
    circleColor: "#F29D4B",
    circleRadius: 5,
    circleStrokeColor: "#fff",
    circleStrokeWidth: 2
  }
});

type ObservationFeature = {
  type: "Feature",
  geometry: {
    type: "Point",
    coordinates: [number, number] | [number, number, number]
  },
  properties: {| id: string |}
};

/**
 * Convert a map of observations into a GeoJSON FeatureCollection
 *
 * @param {{ [string]: ObservationType }} Observations
 * @returns GeoJSON FeatureCollection with Features that have the observation
 * location and id
 */
function mapObservationsToFeatures(
  obsMap: ObservationsMap
): ObservationFeature[] {
  const features = [];
  for (let obs of obsMap.values()) {
    // Only include observations with a location in the map view
    if (
      typeof obs.value.lon === "number" &&
      typeof obs.value.lat === "number"
    ) {
      features.push({
        type: "Feature",
        geometry: {
          type: "Point",
          coordinates: [obs.value.lon, obs.value.lat]
        },
        properties: {
          id: obs.id
        }
      });
    }
  }
  return features;
}

class ObservationMapLayer extends React.PureComponent<{
  onPress: Function,
  observations: ObservationsMap
}> {
  render() {
    const { onPress, observations } = this.props;
    const featureCollection = {
      type: "FeatureCollection",
      features: mapObservationsToFeatures(observations)
    };
    return (
      <MapboxGL.ShapeSource
        onPress={onPress}
        id={`observations-source`}
        shape={featureCollection}
      >
        <MapboxGL.CircleLayer id={`circles`} style={mapboxStyles.observation} />
      </MapboxGL.ShapeSource>
    );
  }
}

type Props = {
  observations: ObservationsMap,
  mapStyle?: MapStyle,
  onAddPress: () => void,
  onPressObservation: (observationId: string) => void
};

class MapView extends React.Component<Props> {
  static defaultProps = {
    onAddPress: () => {},
    onPressObservation: () => {}
  };

  constructor(props: Props) {
    super(props);
    MapboxGL.setAccessToken(
      "pk.eyJ1IjoiZ21hY2xlbm5hbiIsImEiOiJSaWVtd2lRIn0.ASYMZE2HhwkAw4Vt7SavEg"
    );
    MapboxGL.setTelemetryEnabled(false);
    log("accessToken set");
  }

  map: any;

  handleObservationPress = (e: {
    nativeEvent?: {
      payload?: {
        properties?: { id: string }
      }
    }
  }) => {
    log("handle obs press");
    const pressedFeature = e.nativeEvent && e.nativeEvent.payload;
    if (!pressedFeature || !pressedFeature.properties) return;

    const observationId = pressedFeature.properties.id;
    const { observations, onPressObservation } = this.props;
    if (observations.get(observationId)) {
      onPressObservation(observationId);
    } else {
      log(
        "Warning: pressed feature with id '" +
          observationId +
          "' but could not find matching observation"
      );
    }
  };

  handleMapViewRef = (c: any) => {
    this.map = c;
  };

  render() {
    const { observations, onAddPress } = this.props;

    return (
      <>
        <MapboxGL.MapView
          style={{ flex: 1 }}
          ref={this.handleMapViewRef}
          maxZoomLevel={22}
          logoEnabled
          pitchEnabled={false}
          rotateEnabled={false}
          onPress={this.handleObservationPress}
          compassEnabled={false}
          styleURL="mapbox://styles/mapbox/outdoors-v9"
        >
          <ObservationMapLayer
            onPress={this.handleObservationPress}
            observations={observations}
          />
        </MapboxGL.MapView>
        <AddButton onPress={onAddPress} />
      </>
    );
  }
}

export default MapView;
