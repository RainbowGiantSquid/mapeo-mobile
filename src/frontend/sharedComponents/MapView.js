// @flow
import React from "react";
import { View, StyleSheet } from "react-native";
import MapboxGL from "@react-native-mapbox/maps";
import debug from "debug";

// import type { MapStyle } from "../types";
import AddButton from "./AddButton";
import { LocationFollowingIcon, LocationNoFollowIcon } from "./icons";
import IconButton from "./IconButton";
import type {
  LocationContextType,
  PositionType
} from "../context/LocationContext";
import type { ObservationsMap } from "../context/ObservationsContext";

const log = debug("mapeo:MapView");

const mapboxStyles = {
  observation: {
    circleColor: "#F29D4B",
    circleRadius: 5,
    circleStrokeColor: "#fff",
    circleStrokeWidth: 2
  }
};

type ObservationFeature = {
  type: "Feature",
  geometry: {
    type: "Point",
    coordinates: [number, number] | [number, number, number]
  },
  properties: {| id: string |}
};

/**
 * Convert a map of observations into a GeoJSON FeatureCollection
 *
 * @param {{ [string]: ObservationType }} Observations
 * @returns GeoJSON FeatureCollection with Features that have the observation
 * location and id
 */
function mapObservationsToFeatures(
  obsMap: ObservationsMap
): ObservationFeature[] {
  const features = [];
  for (let obs of obsMap.values()) {
    // Only include observations with a location in the map view
    if (
      typeof obs.value.lon === "number" &&
      typeof obs.value.lat === "number"
    ) {
      features.push({
        type: "Feature",
        geometry: {
          type: "Point",
          coordinates: [obs.value.lon, obs.value.lat]
        },
        properties: {
          id: obs.id
        }
      });
    }
  }
  return features;
}

class ObservationMapLayer extends React.PureComponent<{
  onPress: Function,
  observations: ObservationsMap
}> {
  render() {
    const { onPress, observations } = this.props;
    const featureCollection = {
      type: "FeatureCollection",
      features: mapObservationsToFeatures(observations)
    };
    return (
      <MapboxGL.ShapeSource
        onPress={onPress}
        id={`observations-source`}
        shape={featureCollection}
      >
        <MapboxGL.CircleLayer id={`circles`} style={mapboxStyles.observation} />
      </MapboxGL.ShapeSource>
    );
  }
}

type Props = {
  observations: ObservationsMap,
  styleURL: string,
  location: LocationContextType,
  onAddPress: () => any,
  onPressObservation: (observationId: string) => any
};

type State = {
  // True if the map is following user location
  following: boolean
};

class MapView extends React.Component<Props, State> {
  static defaultProps = {
    onAddPress: () => {},
    onPressObservation: () => {}
  };
  state = { following: true };
  map: any;
  initialPosition: void | null | PositionType;

  constructor(props: Props) {
    super(props);
    MapboxGL.setAccessToken(
      "pk.eyJ1IjoiZ21hY2xlbm5hbiIsImEiOiJSaWVtd2lRIn0.ASYMZE2HhwkAw4Vt7SavEg"
    );
    MapboxGL.setTelemetryEnabled(false);
    log("accessToken set");
    this.initialPosition =
      props.location.position || props.location.savedPosition;
  }

  // We only use the location prop (which contains the app GPS location) for the
  // first render of the map. After that location updates come from the native
  // map view, so we don't want to re-render this component every time there is
  // a GPS update
  shouldComponentUpdate(nextProps: Props, nextState: State) {
    return (
      shallowDiffers(this.props, nextProps, ["location"]) ||
      shallowDiffers(this.state, nextState, ["location"])
    );
  }

  handleObservationPress = (e: {
    nativeEvent?: {
      payload?: {
        properties?: { id: string }
      }
    }
  }) => {
    const pressedFeature = e.nativeEvent && e.nativeEvent.payload;
    if (!pressedFeature || !pressedFeature.properties) return;

    const observationId = pressedFeature.properties.id;
    log("handle obs press", observationId);
    const { observations, onPressObservation } = this.props;
    if (observations.get(observationId)) {
      onPressObservation(observationId);
    } else {
      log(
        "Warning: pressed feature with id '" +
          observationId +
          "' but could not find matching observation"
      );
    }
  };

  handleMapViewRef = (c: any) => {
    this.map = c;
  };

  handleRegionChange = (e: any) => {
    // Any user interaction with the map switches follow mode to false
    if (e.properties.isUserInteraction) this.setState({ following: false });
  };

  handleLocationPress = () => {
    this.setState(state => ({ following: !state.following }));
  };

  render() {
    const { observations, onAddPress, styleURL } = this.props;
    const initialCoords = this.initialPosition
      ? [
          this.initialPosition.coords.longitude,
          this.initialPosition.coords.latitude
        ]
      : [0, 0];
    const initialZoom = this.initialPosition ? 8 : 0;
    return (
      <>
        <MapboxGL.MapView
          style={{ flex: 1 }}
          ref={this.handleMapViewRef}
          maxZoomLevel={22}
          surfaceView
          logoEnabled={false}
          pitchEnabled={false}
          rotateEnabled={false}
          onPress={this.handleObservationPress}
          compassEnabled={false}
          styleURL={styleURL}
          onRegionWillChange={this.handleRegionChange}
          regionWillChangeDebounceTime={200}
        >
          {!this.state.following && (
            <MapboxGL.UserLocation showUserLocation={true} />
          )}
          <MapboxGL.Camera
            centerCoordinate={initialCoords}
            zoomLevel={initialZoom}
            followUserLocation={this.state.following}
            followZoomLevel={12}
          />
          <ObservationMapLayer
            onPress={this.handleObservationPress}
            observations={observations}
          />
        </MapboxGL.MapView>
        <AddButton onPress={onAddPress} />
        <View style={styles.locationButton}>
          <IconButton onPress={this.handleLocationPress}>
            {this.state.following ? (
              <LocationFollowingIcon />
            ) : (
              <LocationNoFollowIcon />
            )}
          </IconButton>
        </View>
      </>
    );
  }
}

export default MapView;

// Shallow compare objects, but omitting certain keys from the comparison
function shallowDiffers(a: any, b: any, omit: string[]) {
  for (let i in a) if (!(i in b)) return true;
  for (let i in b) {
    if (a[i] !== b[i] && omit.indexOf(i) === -1) return true;
  }
  return false;
}

const styles = StyleSheet.create({
  locationButton: {
    position: "absolute",
    right: 20,
    bottom: 20
  }
});
