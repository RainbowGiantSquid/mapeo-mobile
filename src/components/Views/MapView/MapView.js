// @flow
import React from 'react';
import { NavigationActions } from 'react-navigation';
import {
  StyleSheet,
  View,
  TouchableOpacity,
  ActivityIndicator,
  Text,
  Modal,
  Dimensions,
  ImageBackground,
  Image
} from 'react-native';
import Drawer from 'react-native-drawer';
import I18n from 'react-native-i18n';
import Icon from 'react-native-vector-icons/MaterialIcons';
import CollectionsImg from 'react-native-vector-icons/MaterialIcons';
import type { Observation } from '../../../types/observation';
import ObservationsView from '../ObservationsView';
import Map from './Map';
import {
  WHITE,
  MAPEO_BLUE,
  LIGHT_GREY,
  VERY_LIGHT_BLUE,
  MEDIUM_GREY
} from '../../../lib/styles';
import Header from '../../Base/Header';
import SavedModal from '../../Base/SavedModal';

const styles = StyleSheet.create({
  buttonText: {
    alignSelf: 'center',
    fontSize: 18,
    fontWeight: '400',
    color: MAPEO_BLUE
  },
  myObservationsIcon: {
    position: 'absolute',
    right: 20,
    top: 15
  },
  cameraIcon: {
    position: 'absolute',
    left: 20,
    top: 15
  },
  profileIcon: {
    position: 'absolute',
    left: 20,
    top: 15
  }
});

type Props = {
  navigation: NavigationActions
};

export type StateProps = {
  showSavedModal: boolean
};

export type DispatchProps = {
  onDrawerClose: () => void,
  onDrawerOpen: () => void
};

I18n.fallbacks = true;
I18n.translations = {
  en: require('../../../translations/en'),
  es: require('../../../translations/es')
};

class MapView extends React.Component<Props & StateProps & DispatchProps> {
  rightDrawer: Drawer;

  closeRightDrawer = () => {
    this.rightDrawer.close();
  };

  openRightDrawer = () => {
    this.rightDrawer.open();
  };

  handleRightDrawerRef = (ref: Drawer) => {
    this.rightDrawer = ref;
  };

  goToCameraView = () => {
    this.props.navigation.navigate({ routeName: 'CameraView' });
  };

  render() {
    const {
      navigation,
      onDrawerClose,
      onDrawerOpen,
      showSavedModal
    } = this.props;

    return (
      <Drawer
        ref={this.handleRightDrawerRef}
        content={<ObservationsView closeRightDrawer={this.closeRightDrawer} />}
        onCloseStart={onDrawerClose}
        onOpenStart={onDrawerOpen}
        openDrawerOffset={0}
        side="right"
        type="displace"
      >
        <Header
          leftIcon={
            <TouchableOpacity onPress={this.goToCameraView}>
              <Icon color={WHITE} name="photo-camera" size={30} />
            </TouchableOpacity>
          }
          rightIcon={
            <TouchableOpacity onPress={this.openRightDrawer}>
              <CollectionsImg color={WHITE} name="collections" size={30} />
            </TouchableOpacity>
          }
          style={{
            position: 'absolute',
            top: 0,
            left: 0,
            right: 0,
            zIndex: 5
          }}
        />
        {showSavedModal && <SavedModal />}
        <Map />
      </Drawer>
    );
  }
}

export default MapView;
