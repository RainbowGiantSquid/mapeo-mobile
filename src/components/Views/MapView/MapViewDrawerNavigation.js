import { DrawerNavigator } from 'react-navigation';

import MyObservationsView from '@src/components/Views/MyObservationsView/MyObservationsView';
import PreferencesView from '@src/components/Views/PreferencesView/PreferencesView';
import MapView from '@src/components/Views/MapView/MapView';

const myObservationsDrawerRouteConfiguration = {
  MapView: {
    screen: MapView,
  },
};
const myObservationsDrawerNavigatorConfiguration = {
  contentComponent: MyObservationsView,
  drawerPosition: 'right',
  drawerOpenRoute: 'RightDrawerOpen',
};

const MyObservationsDrawer = DrawerNavigator(myObservationsDrawerRouteConfiguration, myObservationsDrawerNavigatorConfiguration);
const MapViewDrawerNavigation = DrawerNavigator({
  Child: {
    screen: MyObservationsDrawer,
    navigationOptions: ({navigation}) => ({
    }),
  }
}, {
  contentComponent: PreferencesView,
});

export default MapViewDrawerNavigation;