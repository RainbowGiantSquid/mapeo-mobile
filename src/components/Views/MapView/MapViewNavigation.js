import React from 'react';
import { Image } from 'react-native';
import MapView from '@src/components/Views/MapView';

class MapViewNavigation extends React.Component {
  
  static navigationOptions = {
    tabBarIcon: ({ tintColor }) => (
      <Image source={require('../../../images/location-arrow.png')} />
    ),
  };

  render() {
    return (
      <MapView />
    );
  }
}

export default MapViewNavigation;