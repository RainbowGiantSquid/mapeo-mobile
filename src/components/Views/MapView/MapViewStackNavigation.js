// @flow
import React from 'react';
import { StackNavigator } from 'react-navigation';
import MapView from '@src/components/Views/MapView';
import NewObservationView from '@src/components/Views/NewObservationView/NewObservationView';

const routeConfiguration = {
  MapView: {
    screen: MapView,
  },
  NewObservationView: {
    screen: NewObservationView,
  }
};
const stackConfiguration = {
  headerMode: 'none',
};

const MapViewStackNavigation = StackNavigator(routeConfiguration, stackConfiguration);

export default MapViewStackNavigation;