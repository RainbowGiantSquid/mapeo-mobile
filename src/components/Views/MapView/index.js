// @flow
import { connect } from 'react-redux';
import { NavigationActions } from 'react-navigation';
import type { Dispatch } from 'redux';
import {
  observationList,
  observationCreate,
  observationUpdate,
  observationSelect
} from '../../../ducks/observations';

import { StoreState } from '../../../types/redux';
import MapView from './MapView';
import type { DispatchProps, StateProps } from './MapView';

const resetAction = NavigationActions.reset({
  index: 0,
  actions: [NavigationActions.navigate({ routeName: 'TabBarNavigation' })]
});

const mapStateToProps = (state: StoreState): StateProps => ({
  observations: state.app.observations
});

function mapDispatchToProps(dispatch: Dispatch): DispatchProps {
  return {
    listObservations: () => dispatch(observationList('')),
    createObservation: observation => dispatch(observationCreate(observation)),
    resetNavigation: () => dispatch(resetAction),
    goToPosition: () =>
      dispatch(NavigationActions.navigate({ routeName: 'Position' })),
    updateObservation: observation => dispatch(observationUpdate(observation)),
    selectObservation: o => dispatch(observationSelect(o)),
    goToObservationDetail: () =>
      dispatch(
        NavigationActions.navigate({ routeName: 'ObservationDetailView' })
      )
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(MapView);
