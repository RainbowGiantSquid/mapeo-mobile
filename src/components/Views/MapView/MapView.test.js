// @flow
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import MapView from './MapView';

jest.mock('../ObservationsView', () => () => null);
jest.mock('./Map', () => () => null);
jest.mock('../../Base/SavedModal', () => () => null);

describe('MapView tests', () => {
  test('snapshots', () => {
    const props = [{ showSavedModal: true }, { showSavedModal: false }];

    let tree;
    props.forEach(p => {
      tree = shallow(<MapView showSavedModal={p.showSavedModal} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
