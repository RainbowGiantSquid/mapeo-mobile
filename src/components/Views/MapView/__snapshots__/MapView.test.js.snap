// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapView tests snapshots 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapView
    showSavedModal={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "acceptDoubleTap": false,
      "acceptPan": true,
      "acceptPanOnDrawer": true,
      "acceptTap": false,
      "captureGestures": "open",
      "children": Array [
        <Unknown
          leftIcon={
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#ffffff"
                name="photo-camera"
                size={30}
              />
            </TouchableOpacity>
          }
          rightIcon={
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#ffffff"
                name="collections"
                size={30}
              />
            </TouchableOpacity>
          }
          style={
            Object {
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 5,
            }
          }
        />,
        <Unknown />,
        <Unknown />,
      ],
      "closedDrawerOffset": 0,
      "content": <Unknown
        closeRightDrawer={[Function]}
      />,
      "disabled": false,
      "elevation": 0,
      "initializeOpen": false,
      "negotiatePan": false,
      "onClose": [Function],
      "onCloseStart": undefined,
      "onOpen": [Function],
      "onOpenStart": undefined,
      "open": null,
      "openDrawerOffset": 0,
      "panCloseMask": null,
      "panOpenMask": null,
      "panThreshold": 0.25,
      "side": "right",
      "styles": Object {},
      "tapToClose": false,
      "tweenDuration": 250,
      "tweenEasing": "linear",
      "tweenHandler": null,
      "type": "displace",
      "useInteractionManager": false,
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "leftIcon": <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Icon
              allowFontScaling={false}
              color="#ffffff"
              name="photo-camera"
              size={30}
            />
          </TouchableOpacity>,
          "rightIcon": <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Icon
              allowFontScaling={false}
              color="#ffffff"
              name="collections"
              size={30}
            />
          </TouchableOpacity>,
          "style": Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 5,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "acceptDoubleTap": false,
        "acceptPan": true,
        "acceptPanOnDrawer": true,
        "acceptTap": false,
        "captureGestures": "open",
        "children": Array [
          <Unknown
            leftIcon={
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#ffffff"
                  name="photo-camera"
                  size={30}
                />
              </TouchableOpacity>
            }
            rightIcon={
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#ffffff"
                  name="collections"
                  size={30}
                />
              </TouchableOpacity>
            }
            style={
              Object {
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 5,
              }
            }
          />,
          <Unknown />,
          <Unknown />,
        ],
        "closedDrawerOffset": 0,
        "content": <Unknown
          closeRightDrawer={[Function]}
        />,
        "disabled": false,
        "elevation": 0,
        "initializeOpen": false,
        "negotiatePan": false,
        "onClose": [Function],
        "onCloseStart": undefined,
        "onOpen": [Function],
        "onOpenStart": undefined,
        "open": null,
        "openDrawerOffset": 0,
        "panCloseMask": null,
        "panOpenMask": null,
        "panThreshold": 0.25,
        "side": "right",
        "styles": Object {},
        "tapToClose": false,
        "tweenDuration": 250,
        "tweenEasing": "linear",
        "tweenHandler": null,
        "type": "displace",
        "useInteractionManager": false,
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "leftIcon": <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#ffffff"
                name="photo-camera"
                size={30}
              />
            </TouchableOpacity>,
            "rightIcon": <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#ffffff"
                name="collections"
                size={30}
              />
            </TouchableOpacity>,
            "style": Object {
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 5,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MapView tests snapshots 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapView
    showSavedModal={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "acceptDoubleTap": false,
      "acceptPan": true,
      "acceptPanOnDrawer": true,
      "acceptTap": false,
      "captureGestures": "open",
      "children": Array [
        <Unknown
          leftIcon={
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#ffffff"
                name="photo-camera"
                size={30}
              />
            </TouchableOpacity>
          }
          rightIcon={
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#ffffff"
                name="collections"
                size={30}
              />
            </TouchableOpacity>
          }
          style={
            Object {
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 5,
            }
          }
        />,
        false,
        <Unknown />,
      ],
      "closedDrawerOffset": 0,
      "content": <Unknown
        closeRightDrawer={[Function]}
      />,
      "disabled": false,
      "elevation": 0,
      "initializeOpen": false,
      "negotiatePan": false,
      "onClose": [Function],
      "onCloseStart": undefined,
      "onOpen": [Function],
      "onOpenStart": undefined,
      "open": null,
      "openDrawerOffset": 0,
      "panCloseMask": null,
      "panOpenMask": null,
      "panThreshold": 0.25,
      "side": "right",
      "styles": Object {},
      "tapToClose": false,
      "tweenDuration": 250,
      "tweenEasing": "linear",
      "tweenHandler": null,
      "type": "displace",
      "useInteractionManager": false,
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "leftIcon": <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Icon
              allowFontScaling={false}
              color="#ffffff"
              name="photo-camera"
              size={30}
            />
          </TouchableOpacity>,
          "rightIcon": <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Icon
              allowFontScaling={false}
              color="#ffffff"
              name="collections"
              size={30}
            />
          </TouchableOpacity>,
          "style": Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 5,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "acceptDoubleTap": false,
        "acceptPan": true,
        "acceptPanOnDrawer": true,
        "acceptTap": false,
        "captureGestures": "open",
        "children": Array [
          <Unknown
            leftIcon={
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#ffffff"
                  name="photo-camera"
                  size={30}
                />
              </TouchableOpacity>
            }
            rightIcon={
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#ffffff"
                  name="collections"
                  size={30}
                />
              </TouchableOpacity>
            }
            style={
              Object {
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 5,
              }
            }
          />,
          false,
          <Unknown />,
        ],
        "closedDrawerOffset": 0,
        "content": <Unknown
          closeRightDrawer={[Function]}
        />,
        "disabled": false,
        "elevation": 0,
        "initializeOpen": false,
        "negotiatePan": false,
        "onClose": [Function],
        "onCloseStart": undefined,
        "onOpen": [Function],
        "onOpenStart": undefined,
        "open": null,
        "openDrawerOffset": 0,
        "panCloseMask": null,
        "panOpenMask": null,
        "panThreshold": 0.25,
        "side": "right",
        "styles": Object {},
        "tapToClose": false,
        "tweenDuration": 250,
        "tweenEasing": "linear",
        "tweenHandler": null,
        "type": "displace",
        "useInteractionManager": false,
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "leftIcon": <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#ffffff"
                name="photo-camera"
                size={30}
              />
            </TouchableOpacity>,
            "rightIcon": <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#ffffff"
                name="collections"
                size={30}
              />
            </TouchableOpacity>,
            "style": Object {
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 5,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
